addons/udes_sale_stock/tests/test_sale_order.py

```diff
@@ -0,0 +1,108 @@
+from odoo.addons.udes_stock.tests import common
+from odoo import fields
+
+
+class TestSaleOrder(common.BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestSaleOrder, cls).setUpClass()
+        # Unlink the TestGoodsOut pull_push rule as it was causing the route to misbehave
+        # Add in the final step to the route which links customers and the warehouse i.e. Trailer Dispatch rule
+        Rule = cls.env["stock.rule"]
+        Rule.search([("name", "=", "TestGoodsOut")]).unlink()
+        cls.rule_out = Rule.create(
+            {
+                "name": "TestTrailer",
+                "route_id": cls.route_out.id,
+                "picking_type_id": cls.picking_type_trailer_dispatch.id,
+                "location_id": cls.picking_type_trailer_dispatch.default_location_dest_id.id,
+                "location_src_id": cls.picking_type_trailer_dispatch.default_location_src_id.id,
+                "action": "pull",
+                "procure_method": "make_to_order",
+            }
+        )
+
+    def test_all_created_pickings_are_attached_to_sale_order(self):
+        """
+        Create the sale order for a single product. Check all the pickings created due to sale order are attached to the sale order.
+        """
+        Picking = self.env["stock.picking"]
```

Comment:
Suggestion: Put this variable into `setUpClass` as it is being called in 3 of the tests.

So in `setUpClass` you can do: `cls.Picking = cls.env["stock.picking"]`

Then you can remove `Picking = self.env["stock.picking"]` from the three tests.